src/EmulatorCompat.sol

```diff
@@ -239,4 +282,14 @@ library UArchCompat {
     }
 
     function putChar(AccessLogs.Context memory a, uint8 c) internal pure {}
+
+    function uint32Log2(uint32 value) external pure returns (uint32) {
```

Comment:
I think we should implement this as close to the cpp version as possible.
```
static inline uint32 uint32Log2(uint32 v) {
    return 31 - __builtin_clz(v);
}
```
@guidanoli recently implement a `clz` in Solidity: https://github.com/cartesi/dave/blob/feature/refactor-idataprovider/cartesi-rollups/contracts/src/Math.sol#L19
Could ask for his help to make a version for `uint32`.